lib/Parse/ParseDecl.cpp

```diff
@@ -1632,6 +1632,9 @@ Parser::parseAvailabilityMacro(SmallVectorImpl<AvailabilitySpec *> &Specs) {
   if (NameMatch == Map.end())
     return makeParserSuccess(); // No match, it could be a standard platform.
 
+  SyntaxParsingContext VersionRestrictionContext(
+      SyntaxContext, SyntaxKind::AvailabilityVersionRestriction);
+
```

Comment:
What does this do in practice here?

Reply:
I have no idea what I'm doing, but it feels like these parsing contexts associate parsed entities with information about the specific grammar rule that was used to parse them (or rather, the underlying syntactic role). 

This particular one lets Syntax know that `SwiftStdlib 5.3` has the same syntactic role as `iOS 12.4`. Otherwise these entities (along with the entire decl the attribute appears on) get classified as unknown, which breaks things (incl. `-verify-syntax-tree`). (Perhaps these macros should have their own role, but I don't know enough to say for sure (c.f. 84827672).)

Cc @akyrtzi to tell me how terribly wrong I am. 😉

Reply:
Interesting I missed this in the past and now I see where this is defined for the classic platform availability. Thanks for fixing it!

Reply:
\cc @ahoppen

Reply:
LGTM